test.js

```diff
@@ -3,20 +3,80 @@
 const assert = require("assert");
 const process = require("process");
 const execa = require("execa");
+const createTestServer = require("create-test-server");
+const tempy = require("tempy");
+const fs = require("fs");
+const path = require("path");
+const del = require("del");
 
-function exit(err) {
+const testDir = tempy.directory();
```

Comment:
I don't think a temp dir is needed, just `require` the JSON files directly and respond with them via `nock` for the npm URLs.

Reply:
The idea behind this:

I was thinking, we can do test like this

```js
test({
  packageJson: {
    dependencies: {
      foo: '1.0.0'
    }
  },
  response: {
    '/foo': {
      dist-tags: {
        latest: "2.0.0"
      },
    }
  },
  expect: {
    dependencies: {
      foo: {
        ...
      }
    }
  }
})
```

so we can kill `test.json`, we need a place to write `package.json`

Reply:
I'm going to send another PR to do this

Reply:
#39